huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go

```diff
@@ -0,0 +1,847 @@
+				Computed:    true,
+				Description: `The protect objects list.`,
+			},
+			"status": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The firewall status.`,
+			},
+			"support_ipv6": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: `Whether IPv6 is supported.`,
+			},
+			"east_west_firewall_inspection_vpc_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The east-west firewall inspection VPC ID.`,
+			},
+		},
+	}
+}
+
+func firewallFlavorSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"version": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the version of the firewall.`,
+			},
+			"extend_eip_count": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `Specifies the extend EIP number of the firewall.`,
+			},
+			"extend_bandwidth": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `Specifies the extend bandwidth of the firewall.`,
+			},
+			"extend_vpc_count": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `Specifies the extend VPC number of the firewall.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceFirewallCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createFirewall: Create a CFW firewall.
+	var (
+		createFirewallHttpUrl = "v2/{project_id}/firewall"
+		createFirewallProduct = "cfw"
+	)
+	createFirewallClient, err := cfg.NewServiceClient(createFirewallProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CFW client: %s", err)
+	}
+
+	createFirewallPath := createFirewallClient.Endpoint + createFirewallHttpUrl
+	createFirewallPath = strings.ReplaceAll(createFirewallPath, "{project_id}", createFirewallClient.ProjectID)
+
+	createFirewallOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+
+	createFirewallOpt.JSONBody = utils.RemoveNil(buildCreateFirewallBodyParams(d, cfg))
+	createFirewallResp, err := createFirewallClient.Request("POST", createFirewallPath, &createFirewallOpt)
+	if err != nil {
+		return diag.Errorf("error creating Firewall: %s", err)
+	}
+
+	createFirewallRespBody, err := utils.FlattenResponse(createFirewallResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	orderID := utils.PathSearch("order_id", createFirewallRespBody, "")
+	jobID := utils.PathSearch("job_id", createFirewallRespBody, "")
+
+	bssClient, err := cfg.BssV2Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating BSS v2 client: %s", err)
+	}
```

Comment:
Better to remove line 229-232 to line 235

Reply:
sure